code/sorting/src/circle_sort/README.md

```diff
@@ -0,0 +1,54 @@
+# Circle Sort
+Circle sort algorithm is an **inplace sorting algorithm** in which we compare the first element to the last element of the given array, the second element to the second last element, and so on and then split the array in two and recurse until we get pairs of sorted elements, which are then put together to get a sorted array.
+
+## Example
+Circle sort is best explained by drawing concentric circles on an array of integers. 
+
+Now, the elements lying on the same circle (i.e., they are diametrically opposite) are compared and swapped. Then the array is split into two and recurse  until there are no more swaps.
+
+![circle_sort](https://cdncontribute.geeksforgeeks.org/wp-content/uploads/circle_sort_example.png)
+
+> Image credits: geeksforgeeks
+
+## Pseudocode
+
+```int inner_circle (int *a, int *b)
```

Comment:
I disagree with the fact that you have used a concrete language for pseudo code. I prefer pseudo code to be a set of instructions. For example:

```
int inner_circle(iterator a, iterator b)
{
    iterator start = a;
    ...
    if start == end then return 0
    while start < end
    {
        ....
    }
    
    ...
    return swap
}